crates/store/src/server/api.rs

```diff
@@ -485,13 +485,25 @@ impl api_server::Api for StoreApi {
         request: Request<GetAccountProofsRequest>,
     ) -> Result<Response<GetAccountProofsResponse>, Status> {
         let request = request.into_inner();
+        if request.account_ids.len() != request.code_commitments.len() {
```

Comment:
Based on the recent discussion, I think this should be an inequality check (i.e., number of code commitments should be smaller than or equal to the number of IDs).